src/machine/board_arduino_mega2560.go

```diff
@@ -0,0 +1,199 @@
+// +build arduino_mega2560
+
+package machine
+
+import (
+	"device/avr"
+	"runtime/volatile"
+)
+
+// Return the current CPU frequency in hertz.
+func CPUFrequency() uint32 {
+	return 16000000
+}
+
+const (
+	// Special pins
+	AREF Pin = 98 //
+	LED  Pin = 26 // PB7
+
+	// Analog pins (real pin numbers)
+	A0  Pin = 97 // PF0
+	A1  Pin = 96 // PF1
+	A2  Pin = 95 // PF2
+	A3  Pin = 94 // PF3
+	A4  Pin = 93 // PF4
+	A5  Pin = 92 // PF5
+	A6  Pin = 91 // PF6
+	A7  Pin = 90 // PF7
+	A8  Pin = 89 // PK0
+	A9  Pin = 88 // PK1
+	A10 Pin = 87 // PK2
+	A11 Pin = 86 // PK3
+	A12 Pin = 85 // PK4
+	A13 Pin = 84 // PK5
+	A14 Pin = 83 // PK6
+	A15 Pin = 82 // PK7
+
+	// Analog Input
+	ADC0  Pin = 0
+	ADC1  Pin = 1
+	ADC2  Pin = 2
+	ADC3  Pin = 3
+	ADC4  Pin = 4
+	ADC5  Pin = 5
+	ADC6  Pin = 6
+	ADC7  Pin = 7
+	ADC8  Pin = 8
+	ADC9  Pin = 9
+	ADC10 Pin = 10
+	ADC11 Pin = 11
+	ADC12 Pin = 12
+	ADC13 Pin = 13
+	ADC14 Pin = 14
+	ADC15 Pin = 15
+
+	// Digital pins
+	D0  Pin = 2  // PE0
+	D1  Pin = 3  // PE1
+	D2  Pin = 6  // PE6
+	D3  Pin = 7  // PE5
+	D4  Pin = 1  // PG5
+	D5  Pin = 5  // PE3
+	D6  Pin = 15 // PH3
+	D7  Pin = 16 // PH4
+	D8  Pin = 17 // PH5
+	D9  Pin = 18 // PH6
+	D10 Pin = 23 // PB4
+	D11 Pin = 24 // PB5
+	D12 Pin = 25 // PB6
+	D13 Pin = 26 // PB7
+	D14 Pin = 64 // PJ1
+	D15 Pin = 63 // PJ0
+	D16 Pin = 13 // PH1
+	D17 Pin = 12 // PH0
+	D18 Pin = 46 // PD3
+	D19 Pin = 45 // PD2
+	D20 Pin = 44 // PD1
+	D21 Pin = 43 // PD0
+	D22 Pin = 78 // PA0
+	D23 Pin = 77 // PA1
+	D24 Pin = 76 // PA2
+	D25 Pin = 75 // PA3
+	D26 Pin = 74 // PA4
+	D27 Pin = 73 // PA5
+	D28 Pin = 72 // PA6
+	D29 Pin = 71 // PA7
+	D30 Pin = 60 // PC7
+	D31 Pin = 59 // PC6
+	D32 Pin = 58 // PC5
+	D33 Pin = 57 // PC4
+	D34 Pin = 56 // PC3
+	D35 Pin = 55 // PC2
+	D36 Pin = 54 // PC1
+	D37 Pin = 53 // PC0
+	D38 Pin = 50 // PD7
+	D39 Pin = 70 // PG2
+	D40 Pin = 52 // PG1
+	D41 Pin = 51 // PG0
+	D42 Pin = 42 // PL7
+	D43 Pin = 41 // PL6
+	D44 Pin = 40 // PL5
+	D45 Pin = 39 // PL4
+	D46 Pin = 38 // PL3
+	D47 Pin = 37 // PL2
+	D48 Pin = 36 // PL1
+	D49 Pin = 35 // PL0
+	D50 Pin = 22 // PB3
+	D51 Pin = 21 // PB2
+	D52 Pin = 20 // PB1
+	D53 Pin = 19 // PB0
+)
+
+// Configure sets the pin to input or output.
+func (p Pin) Configure(config PinConfig) {
+	register, _, mask := p.getRegisterPortMask()
+	if config.Mode == PinOutput { // set output bit
+		register.SetBits(mask)
+	} else { // configure input: clear output bit
+		register.ClearBits(mask)
+	}
+}
+
+// Get returns the current value of a GPIO pin.
+func (p Pin) Get() bool {
+	_, port, mask := p.getRegisterPortMask()
+	return (port.Get() & mask) > 0
+}
+
+func (p Pin) getPortMask() (*volatile.Register8, uint8) {
+	_, port, mask := p.getRegisterPortMask()
+	return port, mask
+}
+
+// getRegisterPortMask returns the register, port, and mask for the pin
+func (p Pin) getRegisterPortMask() (*volatile.Register8, *volatile.Register8, uint8) {
+	if p <= A0 && p > A8 { // A0-A7
```

Comment:
It would be a lot better to use a `switch` statement here. Also, it appears that the condition here is the opposite to what the comment says it is?

Reply:
I kept it as separate `if` statements because the original I copied as a guide used them (`machine_atmega.go` - before this branch)

Why do you think the comment is wrong?
it's not a subtraction operator in this case:  

```	A0  Pin = 97 // PF0
	A1  Pin = 96 // PF1
	A2  Pin = 95 // PF2
	A3  Pin = 94 // PF3
	A4  Pin = 93 // PF4
	A5  Pin = 92 // PF5
	A6  Pin = 91 // PF6
	A7  Pin = 90 // PF7
	A8  Pin = 89 // PK0
```

I guess I should have used set notation?

Reply:
The comments were intended to be pin ranges not mathematical operations.

Reply:
I'll update the code to use a `switch` statement.  I had thought about that originally but thought there was some reason not to since I hadn't seen one used anywhere (although I didn't look very hard)

Reply:
My bad, I only just noticed that the A0 thru A9 is in descending order. Most of the boards this is ascending.

Either way, my comment about `switch` instead of a whole bunch of `if` statements still stands, if you please.